### PR TITLE
Adding forger_reward_address param

### PR DESCRIPTION
**Pull request by ysibirski**
Adding FORGER_REWARD_ADDRESS as well as validity check + small fixes

### CODE REVIEW

File: env/.env.forger.eon.template [hunk 1]
```diff
@@ -31,6 +31,7 @@ SCNODE_FORGER_ENABLED=true
...
 SCNODE_FORGER_RESTRICT=true
+SCNODE_FORGER_REWARD_ADDRESS=
```
Comment (otoumas) [inline env/.env.forger.eon.template:34]: this should be added to gobi template as well

File: scripts/init.sh [hunk 2]
```diff
@@ -88,6 +88,39 @@ if ! [ -f "${ENV_FILE}" ]; then
...
+        read -rp "Do you confirm this is the wallet address you want to use for your FORGER rewards to be sent to: ${forger_reward_address}? ('yes' or 'no') " reward_address_set_2
+      done
+      if [ "${reward_address_set_2}" = "yes" ]; then
+        forger_reward_address="$(strip_0x "${forger_reward_address}")"
+        if [ "${forger_reward_address}" != "invalid" ]; then
+          sed -i "s/SCNODE_FORGER_REWARD_ADDRESS=.*/SCNODE_FORGER_REWARD_ADDRESS=${forger_reward_address}/g" "${ENV_FILE}"
+        else
+          # Removing env_file since exit script will leave it in incomplete state
+          rm -f "${ENV_FILE}"
+          fn_die "Error: Provided wallet address is in the WRONG format!!! Please re-run the init.sh script again. Exiting ..."
```
Comment (AndrewCoal737) [inline scripts/init.sh]: It would be useful to provide a format example, to help them identifying what's wrong?

File: scripts/init.sh [hunk 3]
```diff
@@ -88,6 +88,39 @@ if ! [ -f "${ENV_FILE}" ]; then
   SCNODE_NET_NODENAME="ext-${role_value}-$((RANDOM % 100000 + 1))" || fn_die "Error: could not set NODE_NAME variable for some reason. Fix it before proceeding any further.  Exiting..."
   sed -i "s/SCNODE_NET_NODENAME=.*/SCNODE_NET_NODENAME=${SCNODE_NET_NODENAME}/g" "${ENV_FILE}"
 
+  # Setting explicit wallet address for forger rewards to be sent to vs default local forger
+  if [ "${role_value}" = "forger" ] && [ "${network_value}" = "eon" ]; then
```
Comment (otoumas) [inline scripts/init.sh]: why are we limiting this to eon network? Can't this be used in gobi as well?

**Review by AndrewCoal737 — APPROVED** [review]
Just a NB comment

File: scripts/init.sh [hunk 4]
```diff
@@ -88,6 +88,39 @@ if ! [ -f "${ENV_FILE}" ]; then
...
+    echo -e "\n\033[1m=== Setting up FORGER rewards address ===\033[0m\n"
+    read -rp "Do you want to provide the wallet address for your FORGER rewards to be sent to(FORGER local wallet address is used by default)? ('yes' or 'no') " reward_address_set
```
Comment (otoumas) [inline scripts/init.sh]: missing whitespace between to(

File: scripts/utils.sh [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 # Functions
 fn_die() {
-  echo -e "\n\033[1m${1}\033[0m\n" >&2
+  echo -e "\n\033[0;31m\033[1m${1}\033[0m\n" >&2
```
Comment (otoumas) [inline scripts/utils.sh]: shouldn't this be something like: 
echo -e "\n\033[0;1;31m${1}\033[0m\n" >&2

**Review by otoumas — APPROVED** [review]
LGTM see comments

File: scripts/utils.sh (hunk 5 above)
Comment (ysibirski) [reply scripts/utils.sh]: This will produce BOLD RED color for echo message under fn_die function :shrug:

Comment (ysibirski) [reply scripts/utils.sh]: Try it 
```
$ echo -e "\n\033[0;31m\033[1mERROR MESSAGE\033[0m\n" >&2
```

File: scripts/init.sh (hunk 3 above)
Comment (ysibirski) [reply scripts/init.sh]: I assumed we do not need this for GOBI. @lander86 what do you think?

File: env/.env.forger.eon.template (hunk 1 above)
Comment (ysibirski) [reply env/.env.forger.eon.template:34]: I assumed we do not need this for GOBI. @lander86 what do you think?

File: scripts/utils.sh [hunk 6]
```diff
@@ -128,3 +128,26 @@ scnode_start_check() {
     fi
   done
 }
+
+# Function to strip 0x prefix
+strip_0x() {
+  local usage="Check validity of etherium wallet address - usage: ${FUNCNAME[0]} {eth_wallet_address}"
```
Comment (cronicc) [inline scripts/utils.sh]:
```suggestion
  local usage="Check validity of Ethereum wallet address - usage: ${FUNCNAME[0]} {eth_wallet_address}"
```

File: scripts/utils.sh [hunk 7]
```diff
@@ -128,3 +128,26 @@ scnode_start_check() {
...
+  }
+
+  local input="${1}"
+  local eth_address_regex="[0-9a-fA-F]{40}$"
+
+  # Check if the input starts with "0x"
+  if [[ "${input}" =~ ^0x${eth_address_regex} ]]; then
+    # Remove the "0x" prefix
+    echo "${input:2}"
+  elif [[ "${input}" =~ ${eth_address_regex} ]]; then
```
Comment (cronicc) [inline scripts/utils.sh]:
```suggestion
  elif [[ "${input}" =~ ^${eth_address_regex} ]]; then
```

**Review by cronicc — APPROVED** [review]

**Pull request merged**